dvc/tree/webhdfs.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import os
+import threading
+
+from funcy import cached_property, wrap_prop
+
+from dvc.path_info import CloudURLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+class WebHDFSTree(BaseTree):
+    scheme = Schemes.WEBHDFS
+    PATH_CLS = CloudURLInfo
+    REQUIRES = {"hdfs": "hdfs"}
+
+    def __init__(self, repo, config):
+        super().__init__(repo, config)
+
+        url = config.get("url")
+        self.path_info = self.PATH_CLS(url) if url else None
+
+        self.hdfscli_config = (
+            config.get("hdfscli_config")
+            or os.getenv("HDFSCLI_CONFIG")
+            or "~/.hdfscli.cfg"
```

Comment:
Is it the same for windows?

Reply:
The library `hdfs` calls `os.path.expanduser` to generate the appropriate default path in case nothing is passed, in particular `os.path.expanduser` takes care of dealing with Windows' paths. However, since my original approach would default to `"~/.hdfscli.cfg"`, `hdfs` never used it's own default path which called `os.path.expanduser`. This would have caused issues on Windows, like you suggested. 

I've removed the fallbacks so that `None` would be passed if nothing is set in `config`. `hdfs` already handles the default path anyways, so they were redundant.